aodndata/soop/soop_xbt_nrt.py

```diff
@@ -1,13 +1,447 @@
+    :param profile:
+    :param url: url of the ANDS xbt line vocabulary
+    :return:
+    """
+
+    helper = XbtLineVocabHelper(url)
+    xbt_lines_info = helper.xbt_line_info()
+    xbt_line = profile['profile_metadata']['XBT_line']
+
+    # look for xbt line value from BUFR file available in ANDS XBT line vocabulary
+    if xbt_line not in xbt_lines_info:
+        raise InvalidFileContentError(
+            '{xbt_code} is not a known/correct XBT line value found in the ANDS XBT line vocabulary'.
+            format(xbt_code=xbt_line))
+
+    xbt_line_description = xbt_lines_info[xbt_line]['xbt_line_description']
+    xbt_line = xbt_lines_info[xbt_line]['xbt_pref_label']
+
+    profile['profile_metadata']['XBT_line'] = xbt_line
+    profile['profile_metadata']['XBT_line_description'] = xbt_line_description
+
+    return profile
+
+
+def fzf_vessel_get_info(profile):
+    """
+    fuzzy search finder for vessel callsign.
+    The vessel name found in the BUFR file format is slightly different from the platform code AODN vocabulary.
+    This function finds the closest match and modifies the name of the vessel accordingly with a confidence of 0.9
+    :param profile:
+    :return: profile
+    """
+    callsign_list = ship_callsign_list()
+    ship_name = profile['profile_metadata']['ship_name']
+    ship_name_fuzzy_search = fwprocess.extractOne(ship_name,
+                                                  list(callsign_list.values()),
+                                                  score_cutoff=90)
     try:
-        rel_path = filepath.split("sbddata/", 1)[1]
-    except IndexError:
-        raise InvalidFileNameError(
-            'invalid file name {filepath}. Subdirectory \'sbddata\' not found in path, unable to split'.format(filepath=filepath))
-    return os.path.join('IMOS/SOOP/SOOP-XBT/REALTIME', rel_path)
+        ship_name = ship_name_fuzzy_search[0]
+    except:
+        raise InvalidFileContentError('{ship_name} is not a valid enough value to fuzzy match it with an existing AODN'
+                                      'vessel name'.format(ship_name=ship_name))
+
+    callsign = next(k for k, v in callsign_list.items() if v == ship_name)  # find callsign from key value
+    # special case for the Astrolabe vessel. All NRT XBT data (post 2020) is collected by the new Astrolabe vessel,
+    # which has a new callsign.
+    # 2 callsigns are available in the ANDS vocabulary for this vessel; We simply force it to use the new vessel only to
+    # avoid writing complicated and unnecessary logic
+    if callsign == 'FHZI':
+        callsign = 'FASB'
+
+    profile['profile_metadata']['ship_name'] = ship_name
+    profile['profile_metadata']['Callsign'] = callsign
+
+    return profile
+
+
+def netcdf_writer(profile_data, output_dir):
+    """
+    Create a NetCDF from a parsed BUFR csv file
+    :param profile_data:
+    :param output_dir:
+    :return:
+    """
+
+    template = DatasetTemplate.from_json(NC_JSON_TEMPLATE_BUFR)
```

Comment:
FYI if you use [`ImosTemplate`](https://github.com/aodn/python-aodntools/blob/master/aodntools/ncwriter/imos_template.py#L14) instead, some of the fixed [standard global attributes](https://github.com/aodn/python-aodntools/blob/master/aodntools/ncwriter/imos_global.json) will be set by default, so you don't need to add them in your own template. Also things like the time_covarage and geospatial_lat/lon ranges can be automatically set with the `add_extent_attributes()`, and the creation date with `add_date_created_attribute()`

Reply:
thanks!